### PR TITLE
Milestone 1: Project Authentication

### PR DESCRIPTION
**Pull request by OlukaDenis**
- Sign in and signout
- Forms
- Sessions
- Validations
- Remember me

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        @current_user = user
+      end
+    end
+  end
+
+  # Returns true if the the user is logged in, false if the user in logged out
+  def logged_in?
+    !current_user.nil?
+  end
+
```
Comment (imhta) [inline app/helpers/sessions_helper.rb:26]: The `current user` in a SessionHelper that is included in the Application Controller. This is ok, however against the project specification instruction. kindly add those to the application controller

**Review by imhta — CHANGES_REQUESTED** [review]
# Changes required 🛑 

### Hello  @OlukaDenis  , 👋 You have done a great job in this project 👏 
> Did you inform the student success team about working alone? If not kindly inform

> Behaivour of the application is fine. **[UPDATED}**
* [ ] You should use Git Flow: production `master` branch + main `development` branch + Pull Requests from feature branches to the `develop` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/git_flow_requirement.png) from Pathwright)
* [ ] It should have a #before_create callback that generates a :remember_token ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_create_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
* [ ] It should have has_many posts *(not explicity said in specification - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_associations_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) but with extra hint - [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/associations_hint_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) )*

##### ApplicationController

* [ ] It should uses the #current_user method to retrive user ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/current_user_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))


* [ ] It should  uses the #current_user=(user) method to set user ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/current_user_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

* [ ] It should  has a sign_in method ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
* [ ] It has a sign_out method ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/session_delete_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
----
### Don't forget to request another code review
----
**Pon Muthu Selvam N**
__nponmuthuselvam@gmail.com__
_If you have any queries, feel free to ping me on slack_

File: app/helpers/sessions_helper.rb (hunk 1 above)
Comment (OlukaDenis) [reply app/helpers/sessions_helper.rb:26]: Okay @imhta

**Comment by OlukaDenis** [comment]
Hey @imhta, It is clearly shown from the project description not create a signup page. I already created a database seed, you could run `rails db:seed` to populate the test users to the database.
signup

Thanks

**Comment by imhta** [comment]
I am sorry @OlukaDenis . I will redo my review on behaviour and update the review changes.

**Pull request closed without merging**